src/events/request/RequestEventHandler.ts

```diff
@@ -24,94 +23,135 @@ export default class RequestEventHandler implements EventHandler<'message'> {
-			} catch ( error ) {
-				this.logger.error( error );
+			if ( BotConfig.request.invalidRequestJql ) {
+				const tickets = this.getTickets( origin.content );
+				if ( !await RequestsUtil.checkTicketValidity( tickets.join( ',' ) ) ) {
+					try {
+						await origin.react( BotConfig.request.invalidTicketEmoji );
+					} catch ( error ) {
+						this.logger.error( error );
+					}
+
+					try {
+						const warning = await origin.channel.send( `${ origin.author }, your request (<${ origin.url }>) contains a ticket that is less than 24 hours old. Please wait until it is at least one day old before making a request.` );
+
+						const timeout = BotConfig.request.warningLifetime;
+						await warning.delete( { timeout } );
+					} catch ( error ) {
+						this.logger.error( error );
+					}
+					return;
+				}
 			}
-		}
 
-		const internalChannelId = this.internalChannels.get( origin.channel.id );
-		const internalChannel = await DiscordUtil.getChannel( internalChannelId );
+			if ( BotConfig.request.waitingEmoji ) {
+				try {
+					await origin.react( BotConfig.request.waitingEmoji );
+				} catch ( error ) {
+					this.logger.error( error );
+				}
+			}
+
+			const internalChannelId = this.internalChannels.get( origin.channel.id );
+			const internalChannel = await DiscordUtil.getChannel( internalChannelId );
+
+			if ( internalChannel && internalChannel instanceof TextChannel ) {
+				const embed = new MessageEmbed()
+					.setColor( RequestsUtil.getEmbedColor() )
+					.setAuthor( origin.author.tag, origin.author.avatarURL() )
+					.setDescription( this.replaceTicketReferencesWithRichLinks( origin.content, regex ) )
+					.addField( 'Go To', `[Message](${ origin.url }) in ${ origin.channel }`, true )
+					.setTimestamp( new Date() );
 
-		if ( internalChannel && internalChannel instanceof TextChannel ) {
-			const embed = new MessageEmbed()
-				.setColor( RequestsUtil.getEmbedColor() )
-				.setAuthor( origin.author.tag, origin.author.avatarURL() )
-				.setDescription( this.replaceTicketReferencesWithRichLinks( origin.content, regex ) )
-				.addField( 'Go To', `[Message](${ origin.url }) in ${ origin.channel }`, true )
-				.setTimestamp( new Date() );
+				const response = BotConfig.request.prependResponseMessage == PrependResponseMessageType.Always
+					? RequestsUtil.getResponseMessage( origin )
+					: '';
 
-			const response = BotConfig.request.prependResponseMessage == PrependResponseMessageType.Always
-				? RequestsUtil.getResponseMessage( origin )
-				: '';
+				const copy = await internalChannel.send( response, embed ) as Message;
 
-			const copy = await internalChannel.send( response, embed ) as Message;
+				if ( BotConfig.request.suggestedEmoji ) {
+					await ReactionsUtil.reactToMessage( copy, [...BotConfig.request.suggestedEmoji] );
+				}
+			}
+		} else if ( !origin.guild.member( origin.author ).permissionsIn( await DiscordUtil.getChannel( BotConfig.request.logChannel ) ).has( 'VIEW_CHANNEL' ) ) {
```

Comment:
Oh, forgot to mention. This is a really smart way to query who's exempt from this rule. Might extract that one into an utiil function and use elsewhere too in the future. (Although it's probably more clean to have a list of roles...)